### PR TITLE
[ICD] Add LIT ICD reference app for tests and certification

### LINKED ISSUE #29397: [ICD] Test Suite test improbable configuration
Using all-cluster-app leads to weird behavior since the ICDManagementCluster is tested without the proper Init since the define `CHIP_CONFIG_ENABLE_ICD_SERVER` is set to 0. 

this was patched directly in the all-cluster-app Init, however some stricter rules should be applied, like If you support the ICDManagementCluster then `CHIP_CONFIG_ENABLE_ICD_SERVER` shall be set to 1 to prevent unwanted, behavior.

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
Adds a reference example app for the LIT ICD device type.
This will allow us to run intergration tests with an example that implements the LIT ICD requirements.

PR only adds the example to the build steps, test will be added  in a follow up PR.

partially fixes #29397

### CODE REVIEW

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter [hunk 1]
```diff
@@ -0,0 +1,1933 @@
...
+  server cluster Identify {
+    ram      attribute identifyTime default = 0x0000;
+    ram      attribute identifyType default = 0x0;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 4;
+
+    handle command Identify;
+  }
+
+  server cluster Groups {
```
Comment (bzbarsky-apple) [inline examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1605]: Why does this have Groups on the root endpoint?

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter [hunk 2]
```diff
@@ -0,0 +1,1933 @@
...
+    callback attribute partsList;
+    callback attribute generatedCommandList;
+    callback attribute acceptedCommandList;
+    callback attribute eventList;
+    callback attribute attributeList;
+    ram      attribute featureMap default = 0;
+    callback attribute clusterRevision default = 1;
+  }
+
+  server cluster Binding {
```
Comment (bzbarsky-apple) [inline examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1639]: Why does this have Binding on the root endpoint?

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter [hunk 3]
```diff
@@ -0,0 +1,1933 @@
...
+    ram      attribute updatePossible default = 1;
+    ram      attribute updateState default = 0;
+    ram      attribute updateStateProgress default = 0;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 1;
+
+    handle command AnnounceOTAProvider;
+  }
+
+  server cluster UnitLocalization {
```
Comment (bzbarsky-apple) [inline examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1696]: Does this app actually implement UnitLocalization?

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter [hunk 4]
```diff
@@ -0,0 +1,1933 @@
...
+    callback attribute generatedCommandList;
+    callback attribute acceptedCommandList;
+    callback attribute eventList;
+    callback attribute attributeList;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 1;
+  }
+
+  server cluster PowerSource {
```
Comment (bzbarsky-apple) [inline examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1705]: Does this actually implement Power Source?

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter [hunk 5]
```diff
@@ -0,0 +1,1933 @@
...
+    callback attribute attributeList;
+    ram      attribute featureMap default = 0x0007;
+    ram      attribute clusterRevision default = 2;
+
+    handle command RegisterClient;
+    handle command RegisterClientResponse;
+    handle command UnregisterClient;
+  }
+
+  server cluster RelativeHumidityMeasurement {
```
Comment (bzbarsky-apple) [inline examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1924]: Why is RelativeHumidityMeasurement on endpoint 0?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter (hunk 1 above)
Comment (mkardous-silabs) [reply examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1605]: same reason for all the comments, copy paste mistake

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter (hunk 2 above)
Comment (mkardous-silabs) [reply examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1639]: same reason for all the comments, copy paste mistake

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter (hunk 3 above)
Comment (mkardous-silabs) [reply examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1696]: same reason for all the comments, copy paste mistake

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter (hunk 4 above)
Comment (mkardous-silabs) [reply examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1705]: same reason for all the comments, copy paste mistake

File: examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter (hunk 5 above)
Comment (mkardous-silabs) [reply examples/lit-icd-app/lit-icd-common/lit-icd-server-app.matter:1924]: same reason for all the comments, copy paste mistake

**Review by bzbarsky-apple — APPROVED** [review]
Seems ok now, though not having any application endpoints is a bit odd.

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**